[DOC] Documentation simulate (#368)
aramis-lab/leaspy: docs/algorithms.md

## Estimate

> ## Simulate

Reviewer: I am wondering to what extend you should try to make the link with how data are simulated in the litterature, equations? do document a bit the black box behind, WDYT? I am not thinking about something super complex, maybe getting inspiration of [this](https://arxiv.org/pdf/2501.08960) p7

This section describes the procedure for simulating new patient data based on a trained Leaspy model and user-defined parameters. The simulation involves the following steps:

**Step 1: Generation of Individual Parameters**
For each simulated patient, individual parameters (tau, xi, and sources) are sampled from normal distributions defined by the model’s mean and standard deviation. These model parameters come from a previously trained Leaspy model.

> **Step 2: Generation of Visit Times**

Reviewer: Or can be provided by the user?

Visit times are generated based on user-specified visit parameters, such as the number of visits, spacing between visits, and follow-up duration. These parameters are provided through a dictionary.

**Step 3: Estimation of Observations**

> The estimate function from Leaspy is used to compute the patient observations at the generated visit times, based on the individual parameters. Gaussian noise (β-noise) is added to the observations to reflect variability.

Reviewer: Gaissian or Beta? Maybe here say that it is logistic associated with beta? For a linear progression I am not sure it makes sens to put a beta distribution

### Prerequisites
To run a simulation, the following are required:
- A trained Leaspy model (see the `fit` function), used for both parameter sampling (step 1) and the estimate function (step 3).
- A dictionary of visit parameters, specifying the number, type, and timing of visits (used in step 2).
- An `AlgorithmSettings` object, configured for simulation and including:
  - The name of the features to simulate.
  - The visit parameter dictionary.

### Running the Task

```python
>>> from leaspy.algo import AlgorithmSettings
>>> visits_params = {
        'patient_nb': 200,
        'visit_type': "regular",
        'regular_visit': 1,
        'first_visit_mean': 0.,
        'first_visit_std': 0.4,
        'time_follow_up_mean': 11,
        'time_follow_up_std': 0.5,
        'distance_visit_mean': 2 / 12,

>         'distance_visit_std': 0.75 / 12,

Reviewer: Could you maybe document that in the setting options? Or put a ref to the automatically generated documentation of the function?

Reviewer 2: If `simulate` takes an instance of `AlgorithmSettings` as input, then it doesn't need to document it further than something like this:

```python
def simulate(settings: AlgorithmSetttings):
    """Simulate stuffs.

    Parameters
    ----------
    settings : AlgorithmSettings
        The settings for the simulation algorithm.
    """
```

The `AlgorithmSettings` will document its own attributes:

```python
class AlgorithmSettings:
    """Class representing the possible settings for an algorithm.

    Attributes
    ---------
    patient_nb : int
        The number of patients to simulate.

    visit_type : str
        The type of visits to simulate.

    ...
    """
```

Does that make sense ?

    }
>>> simulation_settings = AlgorithmSettings(
        "simulation",
        seed=0,
        features=["MDS1_total", "MDS2_total", "MDS3_off_total", 'SCOPA_total', 'MOCA_total', 'REM_total', 
                  'PUTAMEN_R', 'PUTAMEN_L', 'CAUDATE_R', 'CAUDATE_L'],
        visit_parameters=visits_params
    )
>>> simulated_data = leaspy_logistic.simulate(simulation_settings)

> >>> print(simulated_data.data.to_dataframe().set_index(['ID', 'TIME']).head())

Reviewer: Could you add the actually printed code? as in the nutshell example

Author: Ok, I've added the printed code.

```

### Output

The output is a Data object with ID, TIME and simulated values of each feature. 

### Setting options

> Three options for simulating the visit times exist in Leaspy, which can be specified in visit_param dictionary: 

Reviewer 3: ```suggestion
There are three options to simulate the visit times in Leaspy, which can be specified in visit_param dictionary: 
```

- `random`: Visit times and intervals are sampled from normal distributions.
- `regular`: Visits occur at regular intervals, defined by regular_visit. 
- `dataframe`: Custom visit times are provided directly via a DataFrame.

Refer to the docstring for further details.

## Data Generalities
- monotonicity
- NaN 
- number of visits 
- outliers
- not enough patients 
- parameters don't converge 

> - score don't progress

Reviewer: Necessary? Might cause futur conflicts?
